library/patient.inc.php

```diff
@@ -402,11 +402,49 @@ function getInsuranceDataByDate(
     return sqlQuery($sql, array($pid,$date,$type));
 }
 
+//ALB Calculates unallocated patient balance (pre-payments)
+function get_unallocated_patient_balance($pid)
+{
+    $unallocated = '';
+    $query = "SELECT a.session_id, a.pay_total, a.global_amount " .
+        "FROM ar_session AS a " .
+        "WHERE a.patient_id = ? AND " .
+        "a.adjustment_code = 'pre_payment' AND a.closed = 0";
+    $res = sqlStatement($query, array($pid));
+    while ($row = sqlFetchArray($res)) {
+        $total_amt = $row['pay_total'] - $row['global_amount'];
+        $rs = sqlQuery("SELECT sum(pay_amount) AS total_pay_amt FROM ar_activity WHERE session_id = ? AND pid = ? AND deleted IS NULL", array($row['session_id'], $pid));
+        $pay_amount = $rs['total_pay_amt'];
+        $unallocated += ($total_amt - $pay_amount);
+    }
+    return sprintf('%01.2f', $unallocated);
+}
+
+//ALB Added this function
+function getInsuranceNameByDate(
```

Comment:
not seeing this used anywhere

Reply:
The usage is coming in another commit. I have 4 more from him to post. 
I have done very little cleanup  on his work. I will update the top section and put him as an author. I was using the marking to make sure I got all the right stuff. I was going to wait until final draft and after testing before removing all the ALBs